contrib/masterfiles-stage/common.sh

```diff
@@ -108,6 +108,10 @@ git_deploy_refspec() {
   # and https://github.com/cfengine/core/pull/2465#issuecomment-173656475
   git --git-dir="${local_mirrored_repo}" --work-tree="${temp_stage}" checkout -q -f "${2}^0" ||
     error_exit "Failed to checkout '$2' from '${local_mirrored_repo}'"
+  cd "${temp_stage}"
+  git --git-dir="${local_mirrored_repo}" --work-tree="${temp_stage}" submodule sync
```

Comment:
maybe I can remove `--work-tree` then...

Reply:
I tried without `--work-tree` and without `--git-dir` and without changing dirs and none of those worked. All mentioned that they couldn't operate without a work tree. This combination of `cd` and options seems to work well.